src/xcode/ENA/ENA/Source/Developer Menu/Features/DMSubmissionStateViewController.swift

```diff
@@ -0,0 +1,260 @@
+// Corona-Warn-App
+//
+// SAP SE and all other contributors
+// copyright owners license this file to you under the Apache
+// License, Version 2.0 (the "License"); you may not use this
+// file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#if !RELEASE
+
+import ExposureNotification
+import UIKit
+
+protocol DMSubmissionStateViewControllerDelegate: AnyObject {
+	func submissionStateViewController(
+		_ controller: DMSubmissionStateViewController,
+		getDiagnosisKeys completionHandler: @escaping ENGetDiagnosisKeysHandler
+	)
+}
+
+/// This controller allows you to check if a previous submission of keys successfully ended up in the backend.
+final class DMSubmissionStateViewController: UITableViewController {
+	init(
+		client: Client,
+		delegate: DMSubmissionStateViewControllerDelegate
+	) {
+		self.client = client
+		self.delegate = delegate
+		super.init(style: .plain)
+	}
+
+	@available(*, unavailable)
+	required init?(coder _: NSCoder) {
+		fatalError("init(coder:) has not been implemented")
+	}
+
+	// MARK: UIViewController
+
+	override func viewDidLoad() {
+		super.viewDidLoad()
+		tableView.registerKeyCell()
+	}
+
+	override func viewWillAppear(_ animated: Bool) {
+		navigationController?.setToolbarHidden(false, animated: animated)
+		setToolbarItems(
+			[
+				UIBarButtonItem(
+					barButtonSystemItem: .flexibleSpace,
+					target: nil,
+					action: nil
+				),
+				UIBarButtonItem(
+					title: "Start Test",
+					style: .plain,
+					target: self,
+					action: #selector(performCheck)
+				),
+				UIBarButtonItem(
+					barButtonSystemItem: .flexibleSpace,
+					target: nil,
+					action: nil
+				)
+			],
+			animated: animated
+		)
+		super.viewWillAppear(animated)
+	}
+
+	// MARK: Properties
+
+	private weak var delegate: DMSubmissionStateViewControllerDelegate?
+	private let client: Client
+	private var checkResult = DMSubmittedKeysCheckResult(missingKeys: [], foundKeys: [])
+
+	// MARK: UIViewController
+	
+	@objc
+	func performCheck() {
+		delegate?.submissionStateViewController(self, getDiagnosisKeys: { localKeys, error in
+			if let error = error {
+				fatalError("err: \(error.localizedDescription)")
+			}
+			guard let localKeys = localKeys else {
+				fatalError("err")
+			}
+			print(localKeys)
+			self.client.fetchAllKeys { downloadedPackages in
+				print(downloadedPackages)
```

Comment:
is this print still supposed to be here?

Reply:
done